packages/edit-site/src/components/global-styles/ui.js

```diff
@@ -276,120 +277,158 @@ function GlobalStylesEditorCanvasContainerLink() {
 	}, [ editorCanvasContainerView, isRevisionsOpen, goTo ] );
 }
 
-function GlobalStylesUI() {
+function NavigationSync( { path: parentPath, onPathChange, children } ) {
+	const navigator = useNavigator();
+	const { path: childPath } = navigator.location;
+	const previousParentPath = usePrevious( parentPath );
+	const previousChildPath = usePrevious( childPath );
+	useEffect( () => {
+		console.log({
+			parentPath,
+			previousChildPath,
+			previousParentPath,
+			childPath,
+		});
+		if ( parentPath !== childPath ) {
+			if ( parentPath !== previousParentPath ) {
+				navigator.goTo( parentPath );
+			} else if (
+				childPath !== previousChildPath &&
+				parentPath !== childPath
+			) {
+				onPathChange( childPath );
+			}
+		}
+	}, [
+		onPathChange,
+		parentPath,
+		previousChildPath,
+		previousParentPath,
+		childPath,
+		navigator,
+	] );
+	return children;
+}
+
+function GlobalStylesUI( { path, onPathChange } ) {
```

Comment:
My understanding is that this change is making the `GlobalStylesUI` component controlled. Also by reading the code, it seems that it supports both controlled and uncontrolled modes. I think that's a good approach. Just trying to put it in words to clarify :)

Reply:
That's exactly it the component supports controlled and uncontrolled modes.